### DISCUSSION TITLE
CoupledValueFunctionIC coupling more variables?

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#21657] [Q&A Modules: Phase field] [answered]
Hello,

In a phase-field KKS model, to define the IC of the global concentration variable as sum(c_i*eta_i) where i is phase and c_i is the phase concentration, we can use CoupledValueFunctionIC. This IC can couple variables up to four (x,y,z,t). However, I have a KKS model with more than four phase, so I need to couple more than four etas. Is there any existing method that does similar things as CoupledValueFunctionIC but can couple more variables? Thanks. @dschwen

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont know about more than four variables in this IC since I didnt know about this IC. 
What you could (maybe) do is leverage block restriction if you can make sure that over each subdomain you only need four phases. Then you may use multiple of these CoupledValueFunctionICs with different block restrictions. 

Guillaume

**Reply by xueyang94** [reply]
In general, how does the diffused interface work between two subdomains?

**Reply by GiudGiud** [reply]
as long as you dont block-restrict variables or kernels, only the IC, then the subdomain will have absolutely no impact on the simulation. 

In other terms,
If all objects are block-restricted to two domains, then it is just as if these two blocks were the same

**Reply by xueyang94** [reply]
> 

The variable I want to use this IC for is interpolated within the whole domain, I don't know how using a subdomain will work...

**Reply by xueyang94** [reply]
> as long as you dont block-restrict variables or kernels, only the IC, then the subdomain will have absolutely no impact on the simulation.
> 
> In other terms, If all objects are block-restricted to two domains, then it is just as if these two blocks were the same

If a variable exists within the whole domain, and the domain has a few sub-blocks, how is the interfacial value between the two sub-blocks defined in that case?

**Reply by GiudGiud** [reply]
I m thinking subdomains are just for setting the IC subdomain by subdomain

You mean the value on a node shared between two subdomains?
Not sure. it's probably defined by the order in which the ICs are run but I would need to check

**Comment by dschwen** [comment]
@xueyang94, unfortunately there is no such IC yet. I do have a system that could be used to generate a very high dimensional approximation if you have the data. Another option would be to perform a solve in the IC. We've recently been discussing to use a GEM solver for that.

**Reply by xueyang94** [reply]
Currently I can define the IC for eta using BoundingBoxIC, and define the IC for global concentration using NestedBoundingBoxIC. I think that being able to set global concentration IC directly from eta is more accurate. What data do we need for the system you mentioned?
